### ISSUE TITLE
Support for simplitigs / spectrum-preserving string sets

### ISSUE THREAD

**Issue by karel-brinda** [kowallus/mbgc#5]
Hello, is there currently any support in MBGC for simplitigs (spectrum-preserving string sets)?

The main conceptual difference in terms of file formats is that these are sets of strings so sequence names can be omitted (there're typically many sequences so storing headers is likely to be expensive).

I've tried to experiment with the MBGC capabilities in this regard – eg with the files from https://zenodo.org/records/5555253., But unlike genome assemblies, MBGC with simplitigs seems to have a worse performance than `xz -9 -T1`.

**Comment by karel-brinda** [comment]
I used the following example:
```bash
wget https://zenodo.org/records/5555253/files/neisseria_gonorrhoeae__01.tar.xz
tarcat neisseria_gonorrhoeae__01.tar.xz  | awk '{print ">"NR"\n"$0}' > ng01.fa
mbgc c -m3 -i ng01.fa  ng01.mbgc
```

and got:
```
-rw-r--r-- 1 kbrinda  62M Nov 28 15:40 neisseria_gonorrhoeae__01.tar.xz
-rw-r--r-- 1 kbrinda 921M Nov 28 15:43 ng01.fa
-rw-r--r-- 1 kbrinda  88M Nov 28 15:47 ng01.mbgc
```

**Comment by kowallus** [comment]
No, there is no support for simplitigs. I guess that such extremely short strings are not ideal input for MBGC. 

On the other hand, I would argue about MBGC performance. On my laptop `mbgc c -m3` is >20x faster `xz -9 -T1` losing ~3% in terms of ratio. For example, 'xz -1 -T16' produces over twice as large archive in the same time as MBGC default mode.
We avoided strong backend compression of MBGC streams not to hamper the speed.

**Comment by karel-brinda** [comment]
Thanks for the comments. Just out of interest, how difficult / easy would it be to add such support (essentially fasta -> txt, named -> unnamed sequences)?

In this specific case, it's simplitigs after k-mer propagation. In other protocols, we also do simplitigs without k-mer propagation, and those are longer and would profit from exactly the type of compression MBGC can do.

**Comment by kowallus** [comment]
It should be easy in the case of the max mode or multiple input files. In other cases, to provide parallelism and similar performance more careful changes in implementation might be required. Still, it should not be difficult (only more time-consuming).
